commit 8e8c4d4ae794b96a332c2f0203417b4e32385a64 (refs/changes/04/2604/5)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2017-09-20T14:46:07+00:00 (2 years, 1 month ago)
    
    CAPI-535 runtests is incorrectly reporting success when test suite exits with non zero value
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/test/runtests b/test/runtests
index 23e3359..6b9e1dd 100755
--- a/test/runtests
+++ b/test/runtests
@@ -18,10 +18,9 @@ if [ "$TRACE" != "" ]; then
     export PS4='${BASH_SOURCE}:${LINENO}: ${FUNCNAME[0]:+${FUNCNAME[0]}(): }'
     set -o xtrace
 fi
-# With this stuff enabled we'll get not summary at the end of the tests
-# execution if any test wrote to stderr.
-#set -o errexit
-#set -o pipefail
+
+set -o errexit
+set -o pipefail
 
 
 
@@ -112,10 +111,13 @@ if [[ -n "$opt_test_pattern" ]]; then
     echo "# Running filtered set of test files: $test_files"
 fi
 
+set +o errexit
 if [[ -n "$test_files" ]]; then
     PATH=$NODE_INSTALL/bin:$PATH TEST_CONFIG_FILE=$TOP/etc/config.json node test/test.js \
         | tee $OUTPUT_DIR/ufds.tap
 fi
+rv=$?
+set -o errexit
 
 echo ""
 echo "# test output:"
@@ -147,3 +149,4 @@ echo ""
 if [[ ${tests} != ${passed} ]]; then
     exit 1
 fi
+exit $rv
